src/live-programming.rst

```diff
@@ -36,8 +36,150 @@ Direct Manipulation Programming
+Computer games are not just for fun. They are also being used for training and discovery in many areas including software development, entertainment, education, military, and medicine.
+Game development involves multiple disciplines including programming, design, art, modeling, animation, and music.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'blow2004game'
+
+  This article presents a broad summary of the reasons why game development is hard nowadays.
+  At a high level, the author classified the difficulties into two categories: problems due to overall project size and complexity and problems due to highly domain-specific requirements.
+  Problems due to overall project size and complexity include the lack of excellent development tools, the inefficiency of development workflow, the barriers for multi-platform development, and the lack of excellent third-party components/engines.
+  Problems due to highly domain-specific knowledge requirements include the domain-specific knowledge about programming, mathematics, algorithms, and computer graphics.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'washburn2016went'
+
+  This paper presents an analysis of 155 postmortems published on the gaming site Gamasutra.com. 
+  The paper identifies characteristics of game development, links the characteristics to positive and negative experiences in the postmortems, and distills a set of best practices and pitfalls for game development.
+  Based on the analysis, the authors made the following recommendations to game developers: (1) Be sure to practice good risk management techniques;
+  (2) Prescribe to an iterative development process and utilize prototypes as a method of proving features and concepts before committing them to the design;
+  (3) Do not be overly ambitious in the design.
+
+As we can see in the above paper, there is a huge problem space in the game development domain including product (e.g. art, creativity, game design), development (e.g. tools, development process, team), resources (budge, hardware, publishers), and customer facing (e.g. marketing, feedback).
+This compendium covers only **tools** and **languages** for game development.
+
+Professional Platforms for Game Development
+-------------------------------------------------
+In recent years, game engines such as Unity3D and Unreal Engine have been the most popular platforms for game development.
+A game engine simplifies the task of the programmer by offering convenient abstractions for the hardware and operating systems on top of which the game runs.
+Game engines also provide reusable components such as physics, game inputs, rendering, and artificial intelligence (AI).
+Entity-component systems and event-handling mechanisms have been a fundamental feature of the software architecture of game engines nowadays.
```

Comment:
It is not clear what an "entity-component system" is here -- make sure you define it before you use it.